apps/meteor/tests/end-to-end/api/09-rooms.js

```diff
@@ -1707,4 +1707,230 @@ describe('[Rooms]', function () {
 				.end(done);
 		});
 	});
+	describe('/rooms.isMember', async () => {
+		let testChannel;
+		let testGroup;
+		let testDM;
+
+		const fakeRoomId = `room.test.${Date.now()}-${Math.random()}`;
+		const fakeUserId = `user.test.${Date.now()}-${Math.random()}`;
+
+		const testChannelName = `channel.test.${Date.now()}-${Math.random()}`;
+		const testGroupName = `group.test.${Date.now()}-${Math.random()}`;
+
+		let testUser1;
+		let testUser2;
+		let testUserNonMember;
+		let testUser1Credentials;
+		let testUserNonMemberCredentials;
+
+		it('create users', async () => {
+			[testUser1, testUser2, testUserNonMember] = await Promise.all([createUser(), createUser(), createUser()]);
+			[testUser1Credentials, testUserNonMemberCredentials] = await Promise.all([
+				login(testUser1.username, password),
+				login(testUserNonMember.username, password),
+			]);
+		});
+
+		it('create a channel', (done) => {
+			createRoom({
+				type: 'c',
+				name: testChannelName,
+				members: [testUser1.username, testUser2.username],
+			}).end((err, res) => {
+				testChannel = res.body.channel;
+				done();
+			});
+		});
+		it('create a group', (done) => {
+			createRoom({
+				type: 'p',
+				name: testGroupName,
+				members: [testUser1.username, testUser2.username],
+			}).end((err, res) => {
+				testGroup = res.body.group;
+				done();
+			});
+		});
+		it('create a direct message room', (done) => {
+			createRoom({
+				type: 'd',
+				username: testUser2.username,
+				credentials: testUser1Credentials,
+			}).end((err, res) => {
+				testDM = res.body.room;
+				done();
+			});
+		});
+		after(async () => {
+			await Promise.all([
+				closeRoom({ type: 'c', roomId: testChannel._id }),
+				closeRoom({ type: 'p', roomId: testGroup._id }),
+				closeRoom({ type: 'd', roomId: testDM._id }),
+				deleteUser(testUser1),
+				deleteUser(testUser2),
+				deleteUser(testUserNonMember),
+			]);
+		});
+
+		it('should return error if room not found', (done) => {
+			request
+				.get(api('rooms.isMember'))
+				.set(testUser1Credentials)
+				.query({
+					roomId: fakeRoomId,
+					userId: testUser1._id,
+				})
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('error', 'error-room-not-found');
+				})
+				.end(done);
+		});
+
+		it('should return error if user not found', (done) => {
+			request
+				.get(api('rooms.isMember'))
+				.set(testUser1Credentials)
+				.query({
+					roomId: testChannel._id,
+					userId: fakeUserId,
+				})
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('error', 'error-user-not-found');
+				})
+				.end(done);
+		});
```

Comment:
That's why I requested to remove `roomName` and `username` from the accepted parameters, otherwise, we need to include those test cases here as well, since we are only validating for invalid `roomId` and `userId` here.